x-pack/plugin/ent-search/qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/xpack/application/FullClusterRestartIT.java

```diff
@@ -105,13 +118,36 @@ private void assertDslDataRetention(String analyticsCollectionName) throws IOExc
         for (Object dataStreamObj : dataStreams) {
             ObjectPath dataStream = new ObjectPath(dataStreamObj);
             if (dataStreamName.equals(dataStream.evaluate("name"))) {
-                assertNull(dataStream.evaluate("ilm_policy"));
                 assertEquals(true, dataStream.evaluate("lifecycle.enabled"));
                 assertEquals("180d", dataStream.evaluate("lifecycle.data_retention"));
+                assertEquals("Data stream lifecycle", dataStream.evaluate("next_generation_managed_by"));
+                assertEquals(false, dataStream.evaluate("prefer_ilm"));
```

Comment:
I guess these checks are better indicators than `ilm_policy` being null?

Reply:
The test was not correct and in fact `ilm_policy` is not null but the policy is not used since `prefer_ilm` is `false`.

We was not able to see it before since the test were not executed.

My plan is to open a tech debt issue to check is we can completely remove the ILM policy since it is not used anymore.

Reply:
I have never been fully convince that we have to keep it.
I mean
- Clusters created before 8.12 will have the ILM policy created before they update and data streams created before the update will use it. 
- The registry update does not cleanup useless policies